modules/core/src/main/scala/org/scalasteward/core/vcs/package.scala

```diff
@@ -73,51 +73,90 @@ package object vcs {
     possibleFilenames(baseNames)
   }
 
-  def possibleCompareUrls(repoUrl: Uri, update: Update): List[VersionDiff] = {
+  private[this] def extractRepoVCSType(
```

Comment:
one more, related to the first comment:
`extractRepoVCSType` could live in `VCSExtraAlg` and you would only need to pass the `RepoVCSType`.

Reply:
then `possibleCompareUrls` and `possibleReleaseRelatedUrls` would need to be extracted to `VCSExtraAlg` as well (right now it's in package object), which make sense for me as well

I can try to make it as separated PR

Reply:
@mzuehlke have it as separated PR here: https://github.com/eugeniyk/scala-steward/pull/1
I think we can merge #1563 as is and continue with refactorings in PR above, wdyt?